components/brave_wallet/browser/zcash/rust/unauthorized_orchard_bundle_impl.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_wallet/browser/zcash/rust/unauthorized_orchard_bundle_impl.h"
+
+#include "base/check_is_test.h"
+#include "base/logging.h"
+#include "base/memory/ptr_util.h"
+#include "brave/components/brave_wallet/browser/zcash/rust/authorized_orchard_bundle_impl.h"
+
+namespace brave_wallet::orchard {
+
+UnauthorizedOrchardBundleImpl::UnauthorizedOrchardBundleImpl(
+    ::rust::Box<orchard::OrchardUnauthorizedBundle> orchard_unauthorized_bundle)
+    : orchard_unauthorized_bundle_(std::move(orchard_unauthorized_bundle)) {}
+
+UnauthorizedOrchardBundleImpl::~UnauthorizedOrchardBundleImpl() = default;
+
+// static
+std::unique_ptr<UnauthorizedOrchardBundle> UnauthorizedOrchardBundle::Create(
+    base::span<const uint8_t> tree_state,
+    const std::vector<::brave_wallet::OrchardOutput> orchard_outputs,
+    std::optional<size_t> random_seed_for_testing) {
+  ::rust::Vec<orchard::OrchardOutput> outputs;
+  for (const auto& output : orchard_outputs) {
+    outputs.push_back(orchard::OrchardOutput{output.value, output.addr});
+  }
+  if (random_seed_for_testing) {
```

Comment:
this should come from some global static, not through function args

Reply:
Done